src/main/java/com/google/cloud/discotoproto3converter/proto3/ProtoMerger.java

```diff
@@ -79,32 +81,50 @@ private void mergeMessages(Map<String, Message> newMessages, Map<String, Message
       mergeFields(newMessage, oldMessage, newMessages);
 
       // Merge enums
-      Map<String, Message> newEnumsMap = new HashMap<>();
-      for (Message nestedEnum : newMessage.getEnums()) {
-        newEnumsMap.put(nestedEnum.getName(), nestedEnum);
-      }
-      for (Message oldEnum : oldMessage.getEnums()) {
-        Message newEnum = newEnumsMap.get(oldEnum.getName());
-        mergeFields(newEnum, oldEnum, newMessages);
-      }
+      mergeEnums(newMessages, oldMessage, newMessage);
+    }
+  }
+
+  private void mergeEnums(
+      Map<String, Message> newMessages, Message oldMessage, Message newMessage) {
+    Map<String, Message> newEnumsMap = new HashMap<>();
+    for (Message nestedEnum : newMessage.getEnums()) {
+      newEnumsMap.put(nestedEnum.getName(), nestedEnum);
+    }
+    for (Message oldEnum : oldMessage.getEnums()) {
+      Message newEnum = newEnumsMap.get(oldEnum.getName());
+      mergeFields(newEnum, oldEnum, newMessages);
     }
   }
 
   private void mergeFields(
       Message newMessage, Message oldMessage, Map<String, Message> newMessages) {
     // Merge fields
+    Map<String, Field> newFieldsMap =
+        newMessage.getFields().stream().collect(Collectors.toMap(ProtoElement::getName, f -> f));
+
     for (Field oldField : oldMessage.getFields()) {
       // compareTo() will be used by contains() to search for the field, not equals().
       if (!newMessage.getFields().contains(oldField)) {
-        newMessage.getFields().add(copyField(oldField, newMessages));
+        // Copy removed field
+        newMessage.getFields().add(copyField(null, oldField, newMessages));
+      } else {
+        Field newField = newFieldsMap.get(oldField.getName());
+        // Copy missing options if a new field has less options than old field.
```

Comment:
nitty nits: s/less/fewer/; s/new field/newField/ s/old field/oldField/

Reply:
This is a very nitty nit =)